vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java

```diff
@@ -0,0 +1,87 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.xray.facade;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.vividus.jira.JiraFacade;
+import org.vividus.util.json.JsonPathUtils;
+import org.vividus.xray.configuration.XrayExporterOptions;
+import org.vividus.xray.databind.ManualTestCaseSerializer;
+import org.vividus.xray.model.ManualTestCase;
+
+@Component
+public class XrayFacade
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(XrayFacade.class);
+
+    @Autowired private ManualTestCaseSerializer manualTestSerializer;
+    @Autowired private XrayExporterOptions xrayExporterOptions;
+    @Autowired private JiraFacade jiraFacade;
+
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    public void init()
+    {
+        objectMapper = new ObjectMapper()
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                .registerModule(new SimpleModule().addSerializer(ManualTestCase.class, manualTestSerializer));
+    }
+
+    public String createTestCase(TestCaseParameters testCaseParameters) throws IOException
+    {
+        String createTestRequest = objectMapper.writeValueAsString(createManualTest(testCaseParameters));
+        LOGGER.atInfo().addArgument(createTestRequest).log("Creating Test Case: {}");
+        String response = jiraFacade.createIssue(createTestRequest);
+        String issueKey = JsonPathUtils.getData(response, "$.key");
+        LOGGER.atInfo().addArgument(issueKey).log("Test with key {} has been created");
+        return issueKey;
+    }
+
+    public void updateTestCase(String testCaseKey, TestCaseParameters testCaseParameters) throws IOException
+    {
+        String updateTestRequest = objectMapper.writeValueAsString(createManualTest(testCaseParameters));
+        LOGGER.atInfo().addArgument(testCaseKey)
+                       .addArgument(updateTestRequest)
+                       .log("Updating Test Case with ID {}: {}");
+        jiraFacade.updateIssue(testCaseKey, updateTestRequest);
+        LOGGER.atInfo().addArgument(testCaseKey).log("Test with key {} has been updated");
+    }
+
+    private ManualTestCase createManualTest(TestCaseParameters testCaseParameters)
+    {
+        ManualTestCase manualTest = new ManualTestCase();
+        manualTest.setProjectKey(xrayExporterOptions.getProjectKey());
```

Comment:
will we ever have the necessity to create test cases in different projects?

Reply:
don't think so, but are you asking because TestCaseParameters and ManualTestCase are very similar so we can get rid of one of these classes?

Reply:
I just didn't want to clutter serializer with xray options so to make for it available only the data it can use for serializing

Reply:
I think 'projectKey' should be a field of xrayFacade